src/code/audio/code_8019AF00.c

```diff
@@ -1878,90 +1857,1124 @@ const u8 sIsOcarinaSongReserved[OCARINA_SONG_MAX] = {
+         * wall address to act as a buffer. That way, if a new scarecrow spawn song is
+         * rejected, the previous scarecrow spawn song is not overwritten. If the scarecrow
+         * spawn song is accepted, then the notes are copied over to the scarecrow spawn
+         * song address
+         */
+        recordedSong = sTerminaWallSongPtr;
+    }
+
+    recordedSong[sRecordSongPos].pitch = sRecordOcarinaPitch;
+    recordedSong[sRecordSongPos].length = sOcarinaUpdateTaskStart - sOcarinaRecordTaskStart;
+    recordedSong[sRecordSongPos].volume = sRecordOcarinaVolume;
+    recordedSong[sRecordSongPos].vibrato = sRecordOcarinaVibrato;
+    recordedSong[sRecordSongPos].bend = sRecordOcarinaBendIndex;
+    recordedSong[sRecordSongPos].bFlat4Flag = sRecordOcarinaButtonIndex & 0xC0;
+
+    sRecordOcarinaPitch = sCurOcarinaPitch;
+    sRecordOcarinaVolume = sCurOcarinaVolume;
+    sRecordOcarinaVibrato = sCurOcarinaVibrato;
+    sRecordOcarinaBendIndex = sCurOcarinaBendIndex;
+    sRecordOcarinaButtonIndex = sCurOcarinaButtonIndex;
+
+    sRecordSongPos++;
+
+    if ((sRecordSongPos != (ARRAY_COUNT(sScarecrowsLongSongNotes) - 1)) && !isRecordingComplete) {
+        // Continue recording
+        return;
+    }
+
+    // Recording is complete
+
+    i = sRecordSongPos;
+    pitch = OCARINA_PITCH_NONE;
+    while (i != 0 && pitch == OCARINA_PITCH_NONE) {
+        i--;
+        pitch = recordedSong[i].pitch;
+    }
+
+    if (sRecordSongPos != (i + 1)) {
+        sRecordSongPos = i + 2;
+        recordedSong[sRecordSongPos - 1].length = 0;
+    }
+
+    recordedSong[sRecordSongPos].length = 0;
+
+    if (sRecordingState == OCARINA_RECORD_SCARECROW_SPAWN) {
+        if (sOcarinaStaffPlayingPos >= 8) {
+            for (i = 0; i < sRecordSongPos; i++) {
+                recordedSong[i] = recordedSong[i + 1];
+            }
+
+            // Copies Notes from buffer into scarecrows spawn buttons to be tested for acceptance or rejection
+            AudioOcarina_MapSongFromPitchToButton(OCARINA_SONG_TERMINA_WALL, OCARINA_SONG_SCARECROW_SPAWN, 8);
+
+            // Tests to see if the notes from the scarecrow song contain identical
+            // notes within its song to any of the reserved songIndex from 0 to 21
+
+            // Loop through each of the songs
+            for (i = 0; i < OCARINA_SONG_SCARECROW_SPAWN; i++) {
+                // Checks to see if the song is reserved
+                if (sIsOcarinaSongReserved[i]) {
+                    // Loops through all possible starting indices
+                    for (j = 0; (j < 9 - gOcarinaSongButtons[i].numButtons); j++) {
+                        // Loops through the notes of the song
+                        for (k = 0; k < gOcarinaSongButtons[i].numButtons && k + j < 8 &&
+                                    gOcarinaSongButtons[i].buttonIndex[k] ==
+                                        gOcarinaSongButtons[OCARINA_SONG_SCARECROW_SPAWN].buttonIndex[k + j];
```

Comment:
```suggestion
                        for (k = 0; (k < gOcarinaSongButtons[i].numButtons) && (k + j < 8) &&
                                    (gOcarinaSongButtons[i].buttonIndex[k] ==
                                        gOcarinaSongButtons[OCARINA_SONG_SCARECROW_SPAWN].buttonIndex[k + j]);
```
This one's particularly muddled without.